pkg/cache/cache_test.go

```diff
@@ -789,7 +797,97 @@ func CacheTest(createCacheFunc func(config *rest.Config, opts cache.Options) (ca
 					Eventually(out).Should(Receive(Equal(pod)))
 					close(done)
 				})
+				type selectorsTestCase struct {
+					fieldSelectors map[string]string
+					labelSelectors map[string]string
+					expectedPods   []string
+				}
+				DescribeTable("filter informers at list watch with selectors", func(tc selectorsTestCase) {
+					By("creating the cache")
+					builder := cache.BuilderWithOptions(
+						cache.Options{
+							SelectorsByObject: cache.SelectorsByObject{
+								&kcorev1.Pod{}: {
+									client.MatchingLabels(tc.labelSelectors),
+									client.MatchingFields(tc.fieldSelectors),
+								},
+							},
+						},
+					)
+					informer, err := builder(cfg, cache.Options{})
+					Expect(err).NotTo(HaveOccurred())
+
+					By("running the cache and waiting for it to sync")
+					go func() {
+						defer GinkgoRecover()
+						Expect(informer.Start(informerCacheCtx)).To(Succeed())
+					}()
+					Expect(informer.WaitForCacheSync(informerCacheCtx)).NotTo(BeFalse())
+
+					gvk := schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"}
+					sii, err := informer.GetInformerForKind(context.TODO(), gvk)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(sii).NotTo(BeNil())
+					Expect(sii.HasSynced()).To(BeTrue())
+
+					By("adding an event handler listening for object creation which sends the object to a channel")
```

Comment:
Is it important for us to also test the event handling? I would assume that to already be tested elsewhere and that we could test the relevant aspects of the filtering by just doing `LIST` calls and verifying that we do not get non-matching objects back

Reply:
I think it's not I was just looking for code that tests informers, but since List will retrieve stuff from the cache and the cache is filled in with the informers we are good testing with List.